### PR TITLE
Add article: React for _foreigners_

### PR DESCRIPTION
**Pull request by kristm**
@quipper/devs
Wrote something about my react learning experience. It's actually been sitting in my disk for 2 months now, but maybe some may find it useful. 🙇

### CODE REVIEW

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+author: kristm
+comments: true
+image: https://c2.staticflickr.com/4/3294/2298750408_57553092a6_o.jpg
+---
+
+In the latest [developer survey of StackOverflow](http://stackoverflow.com/insights/survey/2017/), React is the most loved framework and Javascript is the most popular language.<sup>[1](#footnote1)</sup>
+I hear the words _New Hotness_ thrown a lot. Surprisingly, former Hotness Coffeescript has now been relegated to the most hated list.
+
+Backbone nostalgia aside, it does make me raise a question why most developers today choose to make their web frontend increasingly complex. Should all client side code be
+obligatorily heavy just because the technology is there? Or the hardware can (presumably) handle it? Probably not. But rightly or wrongly it seems that's where we are right now. So in order to be part of the steamroller instead of the road<sup>[2](#footnote2)</sup> I'll attempt to grok a react/redux app by stripping it to its bare essential.
```
Comment (kjcpaas) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: Should this be in new line?

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+```
+<script type="text/babel">
+var Hello = React.createClass({
+  getInitialState: function() {
+    return { showModal: false }
+  },
+  toggleModal: function() {
+    this.setState({
+      showModal: !!!this.state.showModal
```
Comment (kjcpaas) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: Maybe only 2 `!` are needed here. 😸

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+```
+<script src="https://cdnjs.cloudflare.com/ajax/libs/redux/3.6.0/redux.min.js"></script>
+```
+
+```
+<script type="text/babel">
+  const Hello = React.createClass({
+    getInitialState: function() {
+      return store.getState();
```
Comment (kjcpaas) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: Why do you need this if you are not using `state`?

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown (hunk 1 above)
Comment (kristm) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: 👍

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown (hunk 2 above)
Comment (kristm) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: ah, right. I'm always thinking i need to force a boolean value via 2 `!!`

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown (hunk 3 above)
Comment (kristm) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: I was under the impression that `initialState` was important 😁

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+layout: post
+title: "React/Redux for foreigners"
+date: 2017-05-01 11:00:00
+author: kristm
+comments: true
+image: https://c2.staticflickr.com/4/3294/2298750408_57553092a6_o.jpg
+---
+
+In the latest [developer survey of StackOverflow](http://stackoverflow.com/insights/survey/2017/), React is the most loved framework and Javascript is the most popular language.<sup>[1](#footnote1)</sup>
+I hear the words _New Hotness_ thrown a lot. Surprisingly, former Hotness Coffeescript has now been relegated to the most hated list.
```
Comment (banyan) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: `s/Coffeescript/CoffeeScript`

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+Backbone nostalgia aside, it does make me raise a question why most developers today choose to make their web frontend increasingly complex. Should all client side code be
+obligatorily heavy just because the technology is there? Or the hardware can (presumably) handle it? Probably not. But rightly or wrongly it seems that's where we are right now.<br><br>So in order to be part of the steamroller instead of the road,<sup>[2](#footnote2)</sup> I'll attempt to grok a react/redux app by stripping it to its bare essential.
+
+### ES6 and Babel
+
+Forget about webpack and all the fancy superflous tooling, React at its core is just ES6. And because it's not supported in all browsers yet, we're going to need a transpiler to manage the ES5 translation<sup>[3](#footnote3)</sup>. That would be Babel. You're also not going to be messing with the DOM as far as React is concerned. It has its own virtual DOM<sup>[4](#footnote4)</sup>. So at the very basic, we're going to be using 3 things<sup>[5](#footnote5)</sup>:
```
Comment (banyan) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: `s/superflous/superfluous/`

File: _posts/2017-05-01-react-redux-for-the-uninitiated.markdown [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+  <Hello />,
+   document.getElementById('container'));
+</script>
+```
+
+Okay, so the State seem to be just a big list of variables that represents various ehem, states of the app.  Here we're using the State to represent the visibility of the modal. Any changes in that state will instantly be reflected in our render method. Suffice to say, one of the reason for react's success is its Virtual DOM implementation<sup>[7](#footnote7)</sup>. 
+
+### Redux
+
+Though not in the same developer survey, one of the things that seem to excite a lot of developers (I know) today is Redux. It's basically a souped-up State when React's State is not enough<sup>[8](#footnote8)</sup>. The website tells it better, Redux is a predictable state container for Javascript apps (which has got to be one of the best written product bylines I've seen).
```
Comment (banyan) [inline _posts/2017-05-01-react-redux-for-the-uninitiated.markdown]: `s/Javascript/JavaScript/`

**Comment by kristm** [comment]
ping @quipper/web-devs

**Comment by banyan** [comment]
@kristm 

Sorry for late! 🚀

**Pull request merged**